server/src/main/java/org/opensearch/index/mapper/MappingLookup.java

```diff
@@ -261,6 +262,15 @@ public String getNestedScope(String path) {
         return null;
     }
 
+    /**
+     * If this index contains @timestamp field with Date type, it will return true
+     * @return true or false based on above condition
+     */
+    public boolean containsTimeStampField() {
```

Comment:
Super nit, may is `hasTimeseries()` would be a bit clearer? But no strong argument it is better than current name

Reply:
```containsTimeStampField``` this can clearly suggest if index has @timestmp field.
A non-timeseries based index can have too @timestamo field ? (I think yes, and that's where this name fits more)